cmd/fs-v1.go

```diff
@@ -622,8 +623,15 @@ func (fs *FSObjects) CopyObject(ctx context.Context, srcBucket, srcObject, dstBu
 		fsMetaPath := pathJoin(fs.fsPath, minioMetaBucket, bucketMetaPrefix, srcBucket, srcObject, fs.metaJSONFile)
 		wlk, err := fs.rwPool.Write(fsMetaPath)
 		if err != nil {
-			logger.LogIf(ctx, err)
-			return oi, toObjectErr(err, srcBucket, srcObject)
+			if !errors.Is(err, errFileNotFound) {
+				logger.LogIf(ctx, err)
+				return oi, toObjectErr(err, srcBucket, srcObject)
+			}
+			wlk, err = fs.rwPool.Create(fsMetaPath)
```

Comment:
I am not able to understand the logic here, as per S3 spec, updating object's metadata will create the object ?

Reply:
I believe this will cause the metadata to be replaced after `fsMeta.ReadFrom(ctx, wlk)` with the default data further down.